raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -531,43 +547,64 @@ contract TokenNetwork is Utils {
+        // permit the closing participant to get the total available deposit.
+        non_closing_amount = (
+            non_closing_participant_state.deposit +
+            closing_transferred_amount -
+            non_closing_participant_state.withdrawn_amount -
+            non_closing_transferred_amount
         );
 
-        // To account for cases when participant2 does not provide participant1's balance proof
-        // Therefore, participant1's transferred_amount will be lower than in reality
-        participant1_amount = min(participant1_amount, total_available_deposit);
+        // In case the non_closing_participant does not provide closing_participant's
+        // correct balance proof, the closing_transferred_amount will be lower than in
+        // reality, possibly causing an underflow.
+        // This is why we limit by the total available deposit.
 
-        // To account for cases when participant1 does not provide participant2's balance proof
-        // Therefore, participant2's transferred_amount will be lower than in reality
-        participant1_amount = max(participant1_amount, 0);
+        // In case the closing participant does not provide the non_closing_participant's
+        // correct balance proof, the non_closing_transferred_amount will be lower than in reality.
+        // Limiting by the total available deposit also helps in this case. Even if the nonclosing
+        // participant receives more tokens than it should, it is the closing participant's fault.
+        non_closing_amount = min(non_closing_amount, total_available_deposit);
 
-        // At this point `participant1_amount` is between [0, total_available_deposit],
+        // At this point `closing_amount` is between [0, total_available_deposit],
         // so this is safe.
-        participant2_amount = total_available_deposit - participant1_amount;
+        closing_amount = total_available_deposit - non_closing_amount;
 
         // Handle old balance proofs with a high locked_amount
-        participant1_amount = max(participant1_amount - participant1_locked_amount, 0);
-        participant2_amount = max(participant2_amount - participant2_locked_amount, 0);
+        if (closing_locked_amount <= closing_amount) {
+            closing_amount = closing_amount - closing_locked_amount;
+        }
+        else {
+            closing_locked_amount = closing_amount;
+            closing_amount = 0;
+        }
+
+        if (non_closing_locked_amount <= non_closing_amount) {
+            non_closing_amount = non_closing_amount - non_closing_locked_amount;
+        }
+        else {
+            non_closing_locked_amount = non_closing_amount;
+            non_closing_amount = 0;
+        }
 
-        require(participant1_amount <= total_available_deposit);
-        require(participant2_amount <= total_available_deposit);
+        require(closing_amount <= total_available_deposit);
```

Comment:
I would make this an `assert` also , because it should never happen and it's not testing something from user input. It is testing 
```
non_closing_amount = min(non_closing_amount, total_available_deposit);
closing_amount = total_available_deposit - non_closing_amount;
```